src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -46,7 +44,7 @@ internal sealed class SlidingTextWindow : IDisposable
         private int _basis;                                // Offset of the window relative to the SourceText start.
         private int _offset;                               // Offset from the start of the window.
         private readonly int _textEnd;                     // Absolute end position
-        private char[] _characterWindow;                   // Moveable window of chars from source text
+        private char[]? _characterWindow;                  // Moveable window of chars from source text
```

Comment:
The only reason this is nullable is because `Dispose` clears it. If that isn't considered a good reason to make this nullable because nobody should use this type after calling `Dispose` anyway, I could just use `!` there.

Reply:
Alternatively, I could add a `ThrowIfDispose` method which would be called at the beginning of every method, the method could be conditional for DEBUG to avoid any possible overhead in release builds, and the methods could use `MemberNotNull`

Reply:
I'd be fine with removing the annotation (and not add `RoslynDebug.AssertNotNull(_characterWindow)` assertions). Let's see what the second reviewer thinks.
If we keep the nullable annotation here, let's update the comment to say "(only null after Dispose was called)" or some such.

Reply:
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/api-guidelines/nullability.md states:

> - **DO NOT** be concerned with the validity of annotations after an object has been Dispose'd.  In general in .NET, using an instance after it's been disposed is a violation of its contract, and so similarly we generally (if the object isn't supposed to be used after disposal) don't consider the nullable annotations to be meaningful after disposal.  This means, for example, that if a property is initialized in a ctor and is always non-null until disposal but then may return null after disposal, it should still be annotated as non-nullable.

Reply:
Cool, didn't know this guidelines, will fix it, thanks 😊

Reply:
Fixed.